nipype/interfaces/dtitk/registration.py

```diff
@@ -1,394 +1,430 @@
+    >>> node.inputs.transform = 'im_warp.df.nii'
+    >>> node.cmdline
+    'deformationSymTensor3DVolume -df FD -in im1.nii -interp LEI -out
+     im1_diffeoxfmd.nii -reorient PPD -trans im_warp.df.nii'
+    >>> node.run() # doctest: +SKIP
+    """
+
+    input_spec = DiffeoSymTensor3DVolInputSpec
+    output_spec = DiffeoSymTensor3DVolOutputSpec
+    _cmd = 'deformationSymTensor3DVolume'
 
-class diffeoScalarVolInputSpec(CommandLineInputSpec):
-    in_volume = traits.Str(desc='moving volume', exists=True, mandatory=False,
-                           position=0, argstr="-in %s")
-    in_xfm = traits.Str(desc='transform to apply', exists=True,
-                        mandatory=False,
-                        position=2, argstr="-trans %s")
-    in_target = traits.Str(desc='', exists=True, mandatory=False, position=3,
-                           argstr="-target %s")
-    out_file = traits.Str(desc='', position=1, argstr="-out %s",
-                          name_source="in_volume",
-                          name_template="%s_diffeoxfmd.nii.gz")
-    in_vsize = traits.Str(desc='', exists=True, mandatory=False, position=4,
-                          argstr="-vsize %s")
-    in_flip = traits.Str(desc='', exists=True, mandatory=False, position=5,
-                         argstr="-flip %s")
-    in_type = traits.Str(desc='', exists=True, mandatory=False, position=6,
-                         argstr="-type %s")
-    in_interp = traits.Str(desc='0 trilin, 1 NN', exists=True, mandatory=False,
-                           position=7, argstr="-interp %s")
-
-
-class diffeoScalarVolOutputSpec(TraitedSpec):
-    out_file = traits.File(desc='moved volume', exists=True)
-
-
-class diffeoScalarVolTask(CommandLineDtitk):
+    def _format_arg(self, name, spec, value):
+        if name == 'resampling_type':
+            value = {'forward': 0, 'backward': 1}[value]
+        return super(DiffeoSymTensor3DVol, self)._format_arg(name, spec, value)
+
+
+class DiffeoScalarVolInputSpec(CommandLineInputSpec):
+    in_file = File(desc='moving scalar volume', exists=True,
+                   argstr="-in %s", mandatory=True)
+    out_file = File(desc='output filename',
+                    argstr="-out %s", name_source="in_file",
+                    name_template="%s_diffeoxfmd", keep_extension=True)
+    transform = transform = File(exists=True, argstr="-trans %s",
+                                 mandatory=True, desc='transform to apply')
+    target = File(exists=True, argstr="-target %s", xor=['voxel_size'],
+                  desc='output volume specification read from the target '
+                  'volume if specified')
+    voxel_size = traits.Tuple((traits.Float(), traits.Float(), traits.Float()),
+                              desc='xyz voxel size (superseded by target)',
+                              argstr="-vsize %g %g %g", xor=['target'])
+    flip = traits.Tuple((traits.Int(), traits.Int(), traits.Int()),
+                        argstr="-flip %d %d %d")
```

Comment:
Description? Also, my guess is this is limited to 0 and 1... Maybe it should be `traits.Tuple((traits.Bool, traits.Bool, traits.Bool), ...)`? The `argstr` could stay the same.

I would add a `flip` input in the docstring to make sure that the `argstr` works properly.

Reply:
I'm guessing this is a binary x y z to flip each axis, but I haven't found that documented.

` -flip          type: integer(s)   required: N	default: 0 0 0 `